priam/src/main/java/com/netflix/priam/tuner/JVMOptionsTuner.java

```diff
@@ -0,0 +1,152 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.netflix.priam.tuner;
+
+import com.google.inject.Inject;
+import com.netflix.priam.IConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * This is to tune the jvm.options file introduced in Cassandra 3.x to pass JVM parameters to Cassandra.
+ * It supports configuring GC type (CMS/G1GC) where it automatically activates default properties as provided in
+ * jvm.options file. Note that this will not "add" any GC options.
+ * <p>
+ * Created by aagrawal on 8/23/17.
+ */
+public class JVMOptionsTuner {
+    private static final Logger logger = LoggerFactory.getLogger(JVMOptionsTuner.class);
+    protected final IConfiguration config;
+
+    @Inject
+    public JVMOptionsTuner(IConfiguration config) {
+        this.config = config;
+    }
+
+    /**
+     * Update the JVM options file for cassandra by updating/removing JVM options {@link IConfiguration#getJVMExcludeSet()} && {@link IConfiguration#getJVMUpsertSet()}, configuring GC {@link IConfiguration#getGCType()}etc.
+     *
+     * @param outputFile File name with which this configured JVM options should be written.
+     * @throws Exception when encountered with invalid configured GC type. {@link IConfiguration#getGCType()}
+     */
+    public void updateJVMOptions(final String outputFile) throws Exception {
+        List<String> configuredJVMOptions = updateJVMOptions();
+
+        if (logger.isInfoEnabled())
+        {
+            StringBuffer buffer = new StringBuffer("\n");
+            configuredJVMOptions.stream().forEach(line -> buffer.append(line).append("\n"));
+            logger.info("Updating jvm.options with following values: " + buffer.toString());
+        }
+
+        //Verify we can write to output file and it is not directory.
+        File file = new File(outputFile);
+        if (file.exists() && !file.canWrite()) {
+            throw new Exception("Not enough permissions to write to file: " + outputFile);
+        }
+
+        //Write jvm.options back to override defaults.
+        Files.write(new File(outputFile).toPath(), configuredJVMOptions);
+    }
+
+    /**
+     * Update the JVM options file for cassandra by updating/removing JVM options {@link IConfiguration#getJVMExcludeSet()} && {@link IConfiguration#getJVMUpsertSet()}, configuring GC {@link IConfiguration#getGCType()}etc.
+     *
+     * @return List of Configuration as String after reading the configuration from jvm.options
+     * @throws Exception when encountered with invalid configured GC type. {@link IConfiguration#getGCType()}
+     */
+    protected List<String> updateJVMOptions() throws Exception {
```

Comment:
How is it different from the above function? If it is different, can you come up with more descriptive name